quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/StoragePropertyMappers.java

```diff
@@ -307,18 +315,19 @@ public static PropertyMapper[] getMappers() {
         };
     }
 
-    private static Optional<String> isForceComponentFactoryCache(Optional<String> storage, ConfigSourceInterceptorContext context) {
-        if (storage.isPresent()) {
-            return Optional.of(Boolean.TRUE.toString());
-        }
-
-        return storage;
-    }
-
     private static Optional<String> getAreaStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
         return of(storage.isEmpty() ? "jpa" : "map");
     }
 
+    private static Optional<String> getResourcesVersionSeed(Optional<String> parameterValue, ConfigSourceInterceptorContext context) {
+        if (!parameterValue.isEmpty()) {
+            return parameterValue;
+        }
+        Logger.getLogger(StoragePropertyMappers.class)
+                .warnf("Version seed for deployment state set with a random number. Caution: This can lead to unstable operations when serving resources from the cluster without a sticky loadbalancer or when restarting nodes. Set the '--%s' option with a secret seed to ensure stable operations.", StorageOptions.STORAGE_DEPLOYMENT_STATE_RESOURCES_VERSION_SEED.getKey());
+        return Optional.of(SecretGenerator.getInstance().randomString(10));
```

Comment:
Have you tried running multiple `kc.sh build`? Does the seed survive the builds? I guess, but not entirely sure atm, so want to re-confirm we don't add another one on every build but only at first build. 

Also: currently, the generated seed is used as part of the url path for resources/assets on the frontend. Is this still true for this seed or am I mixing stuff here and this is another thing? If it is the same, the current seed used is a 5-character random string IIRC, should we align?

Reply:
Also thinking about a better approach using a build step that creates the random seed and writes it into the keycloak.conf file instead, so it's easier to just copy the conf for all keycloak instances. but this may be sth for another iteration.

Reply:
It needs not survive rebuilds - the random can be random everytime. If it should survive, it has to be set by this option.